customerio/customerio-ios
# feat: add page rule for inline inapp messages

[Implement page rules with inline messages](https://linear.app/customerio/issue/MBL-314/implement-page-rules-with-inline-messages)

### Problem:
As a marketer I want some inline in-app messages to only be displayed on certain screens of my app. The in-app message does not display inline on the screen that was set in page rules.

### Testing:
For QA, there are three inline views in UIKit Sample app. 
2 on the dashboard screen - `dashboard-announcement` and `dashboard-announcement-code`
1 on custom attributes screen -  `custom-screen`

To avoid any confusion, since same controller `CustomDataViewController` is used for Custom Attributes, Device Attributes,  Profile Attributes so if an in-app message is targeted for element Id `custom-screen` it will show every time a user visits any of the above screens. 

**Steps to test:**
- Using a campaign/broadcast send an inline inapp message with _no page rule_ targeted for one of the element ids given above example dashboard-announcement
- You'll see this inline message on the targeted element id. 
- Now send another inline inapp message _with page rule_ (assume Custom) targeted for another element id example custom-screen
- Note that the inline inapp message targeted with page rule (Custom) and element Id custom-screen will only be shown on custom view controllers like Custom Attributes, Device Attributes, Profile Attributes since they all share common view controller. 



https://github.com/customerio/customerio-ios/assets/91549653/f5c03705-4e1b-4bcd-85ab-c05b419cb37b
